### ISSUE TITLE
The user tagger does not work on users from the same instance

### ISSUE THREAD

**Issue by Lucki** [howdy-tsc/LemmyTools#49]
From my short investigation it looks like the regex expects an `@` which doesn't apply to local users href.

![grafik](https://github.com/howdy-tsc/LemmyTools/assets/1408843/7d165c9f-6853-4a41-870a-ae14f8b205e1)

![grafik](https://github.com/howdy-tsc/LemmyTools/assets/1408843/20d8e063-fe3b-4664-b62a-c21a3f7cb022)

~~~
TypeError: node.href.match(...) is null
    addNotesToLinks moz-extension://6f3c5c6f-f627-4637-99c3-250b4725999c/ Lemmy: LemmyTools.user.js#76:356
    tagUsers moz-extension://6f3c5c6f-f627-4637-99c3-250b4725999c/ Lemmy: LemmyTools.user.js#76:389
    setInterval handler*tagUsers moz-extension://6f3c5c6f-f627-4637-99c3-250b4725999c/ Lemmy: LemmyTools.user.js#76:385
    VM93kzbs2uk6 moz-extension://6f3c5c6f-f627-4637-99c3-250b4725999c/ Lemmy: LemmyTools.user.js#76:1038
    VM93kzbs2uk6 moz-extension://6f3c5c6f-f627-4637-99c3-250b4725999c/ Lemmy: LemmyTools.user.js#76:1185
    VM93kzbs2uk6 moz-extension://6f3c5c6f-f627-4637-99c3-250b4725999c/ Lemmy: LemmyTools.user.js#76:1186
    Ut moz-extension://6f3c5c6f-f627-4637-99c3-250b4725999c/injected-web.js:1
    <anonymous> moz-extension://6f3c5c6f-f627-4637-99c3-250b4725999c/ Lemmy: LemmyTools.user.js#76:1
 
<a class="person-listing d-inline-…tems-baseline text-info" title="@wintermute" href="/u/wintermute" target="_blank" rel="noreferrer">
​
"$EV": Object { onClick: {…}, onDblClick: null, onFocusIn: null, … }
​
accessKey: ""
​
accessKeyLabel: ""
​
assignedSlot: null
​
attributes: NamedNodeMap(5) [ class="person-listing d-inline-flex align-items-baseline text-info hasNote", title="@wintermute", href="/u/wintermute", … ]
​
autocapitalize: ""
​
autofocus: false
​
baseURI: "https://feddit.de/"
​
charset: ""
​
childElementCount: 2
​
childNodes: NodeList [ span, span.userNote.badge ]
​
children: HTMLCollection { 0: span, 1: span.userNote.badge, length: 2 }
​
classList: DOMTokenList(5) [ "person-listing", "d-inline-flex", "align-items-baseline", … ]
​
className: "person-listing d-inline-flex align-items-baseline text-info hasNote"
​
clientHeight: 18
​
clientLeft: 0
​
clientTop: 0
​
clientWidth: 96
​
contentEditable: "inherit"
​
coords: ""
​
dataset: DOMStringMap(0)
​
dir: ""
​
download: ""
​
draggable: true
​
enterKeyHint: ""
​
firstChild: <span>
​
firstElementChild: <span>
​
hash: ""
​
hidden: false
​
host: "feddit.de"
​
hostname: "feddit.de"
​
href: "https://feddit.de/u/wintermute"
​
hreflang: ""
​
id: ""
​
inert: false
​
innerHTML: '<span>@wintermute</span><span class="userNote badge" data-user="undefined">🏷</span>'
​
innerText: "@wintermute\n🏷"
​
inputMode: ""
​
isConnected: true
​
isContentEditable: false
​
lang: ""
​
lastChild: <span class="userNote badge" data-user="undefined">
​
lastElementChild: <span class="userNote badge" data-user="undefined">
​
localName: "a"
​
name: ""
​
namespaceURI: "http://www.w3.org/1999/xhtml"
​
nextElementSibling: null
​
nextSibling: null
​
nodeName: "A"
​
nodeType: 1
​
nodeValue: null
​
nonce: ""
​
offsetHeight: 18
​
offsetLeft: 105
​
offsetParent: <section id="sidebarInfo" class="card border-secondary mb-3">
​
offsetTop: 857
​
offsetWidth: 96
​
onabort: null
​
onanimationcancel: null
​
onanimationend: null
​
onanimationiteration: null
​
onanimationstart: null
​
onauxclick: null
​
onbeforeinput: null
​
onblur: null
​
oncanplay: null
​
oncanplaythrough: null
​
onchange: null
​
onclick: null
​
onclose: null
​
oncontextmenu: null
​
oncopy: null
​
oncuechange: null
​
oncut: null
​
ondblclick: null
​
ondrag: null
​
ondragend: null
​
ondragenter: null
​
ondragexit: null
​
ondragleave: null
​
ondragover: null
​
ondragstart: null
​
ondrop: null
​
ondurationchange: null
​
onemptied: null
​
onended: null
​
onerror: null
​
onfocus: null
​
onformdata: null
​
onfullscreenchange: null
​
onfullscreenerror: null
​
ongotpointercapture: null
​
oninput: null
​
oninvalid: null
​
onkeydown: null
​
onkeypress: null
​
onkeyup: null
​
onload: null
​
onloadeddata: null
​
onloadedmetadata: null
​
onloadstart: null
​
onlostpointercapture: null
​
onmousedown: null
​
onmousemove: null
​
onmouseout: null
​
onmouseover: null
​
onmouseup: null
​
onpaste: null
​
onpause: null
​
onplay: null
​
onplaying: null
​
onpointercancel: null
​
onpointerdown: null
​
onpointerenter: null
​
onpointerleave: null
​
onpointermove: null
​
onpointerout: null
​
onpointerover: null
​
onpointerup: null
​
onprogress: null
​
onratechange: null
​
onreset: null
​
onresize: null
​
onscroll: null
​
onscrollend: null
​
onsecuritypolicyviolation: null
​
onseeked: null
​
onseeking: null
​
onselect: null
​
onselectionchange: null
​
onselectstart: null
​
onslotchange: null
​
onstalled: null
​
onsubmit: null
​
onsuspend: null
​
ontimeupdate: null
​
ontoggle: null
​
ontransitioncancel: null
​
ontransitionend: null
​
ontransitionrun: null
​
ontransitionstart: null
​
onvolumechange: null
​
onwaiting: null
​
onwebkitanimationend: null
​
onwebkitanimationiteration: null
​
onwebkitanimationstart: null
​
onwebkittransitionend: null
​
onwheel: null
​
origin: "https://feddit.de"
​
outerHTML: '<a class="person-listing d-inline-flex align-items-baseline text-info hasNote" title="@wintermute" href="/u/wintermute" target="_blank" rel="noreferrer"><span>@wintermute</span><span class="userNote badge" data-user="undefined">🏷</span></a>'
​
outerText: "@wintermute\n🏷"
​
ownerDocument: HTMLDocument https://feddit.de/
​
parentElement: <li class="list-inline-item">
​
parentNode: <li class="list-inline-item">
​
part: DOMTokenList []
​
password: ""
​
pathname: "/u/wintermute"
​
ping: ""
​
port: ""
​
prefix: null
​
previousElementSibling: null
​
previousSibling: null
​
protocol: "https:"
​
referrerPolicy: ""
​
rel: "noreferrer"
​
relList: DOMTokenList [ "noreferrer" ]
​
rev: ""
​
scrollHeight: 18
​
scrollLeft: 0
​
scrollLeftMax: 0
​
scrollTop: 0
​
scrollTopMax: 0
​
scrollWidth: 96
​
search: ""
​
shadowRoot: null
​
shape: ""
​
slot: ""
​
spellcheck: false
​
style: CSS2Properties(0)
​
tabIndex: 0
​
tagName: "A"
​
target: "_blank"
​
text: "@wintermute🏷"
​
textContent: "@wintermute🏷"
​
title: "@wintermute"
​
translate: true
​
type: ""
​
username: ""
​
<prototype>: HTMLAnchorElementPrototype { toString: toString(), target: Getter & Setter, download: Getter & Setter, … }
 https://feddit.de/u/wintermute Lemmy: LemmyTools.user.js:358:17

~~~

**Comment by howdy-tsc** [comment]
Can confirm. Will look into this. Thank you for the detailed message.

**Comment by howdy-tsc** [comment]
This has been fixed in 2.0.9
